### PR TITLE
Stream rewrite

### PR DESCRIPTION
**Pull request by VibroAxe**
This PR removes the need for sniproxy by using nginx's built in stream handling. It requires extensive testing before we mainline to make sure nothing breaks.

It requires significant updates to 2 main images (ubuntu-nginx / generic) & minor tweaks to monolithic (adding expose 443) and docker-compose (removing sniproxy and remapping 443)

Please collate all discussion / issues here, I have opened a PR on ubuntu-nginx in case there are code specific comments:https://github.com/lancachenet/ubuntu-nginx/pull/2

From a docker compose:
```
git pull
git checkout stream
docker-compose up -d
```
or if you are running base containers, stop sniproxy, then recreate monolithic using #stream and pass port 443 in as well

### CODE REVIEW

File: overlay/etc/nginx/stream-available/10_sni.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+server {
+	listen 443;
+	resolver 8.8.8.8 ipv6=off;
```
Comment (MathewBurnett) [inline overlay/etc/nginx/stream-available/10_sni.conf]: Should this be using the upstream env var? Also should there be a entry script entry to write the value?

**Comment by MathewBurnett** [comment]
Tested the docker-compose branch for stream using a mono - dns pair. I can see https requests coming through the logs for epic, battlenet, stream and origin as expected. Downloads on the big four also seem to work as intended.

**Comment by MathewBurnett** [comment]
I can see the upstream taking effect on the HTTPS requests and the HTTP seem to cache as expected. This change appears to work as expected in my home network.

**Comment by VibroAxe** [comment]
@mintopia @Lepidopterist any comments or happy to merge?

**Comment by mintopia** [comment]
I've had it running here for a week or so with no issues! Happy to merge.

**Pull request merged**